### PR TITLE
Spring boot jpa

### PR DESCRIPTION
**Pull request by forzen123212**
moved to jpa interface + liquibase support

### CODE REVIEW

File: src/main/java/com/example/springfirsthw/service/BookService.java [hunk 1]
```diff
@@ -1,10 +1,17 @@
 package com.example.springfirsthw.service;
 
-import com.example.springfirsthw.domain.Book;
+import com.example.springfirsthw.dto.BookDto;
+import com.example.springfirsthw.dto.CreateBookRequestDto;
 import java.util.List;
 
 public interface BookService {
-    Book save(Book book);
+    BookDto save(CreateBookRequestDto book);
```
Comment (DEz-1996) [inline src/main/java/com/example/springfirsthw/service/BookService.java]: Let's rename book to bookDto for consistency code

File: src/main/java/com/example/springfirsthw/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -1,27 +1,50 @@
...
+        return bookMapper.toBookDto(book);
     }
 
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        book.setId(id);
+        return bookMapper.toBookDto((bookRepository.save(book)));
```
Comment (DEz-1996) [inline src/main/java/com/example/springfirsthw/service/impl/BookServiceImpl.java]: It's good to check if this ID exists to throw an exception with proper message if not.

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                  name: price
+                  type: decimal
+                  constraints:
+                    nullable: false
+              - column:
+                  name: description
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
```
Comment (DEz-1996) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml]: Add empty line)

File: pom.xml [hunk 4]
```diff
@@ -73,9 +108,43 @@
...
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-compiler-plugin</artifactId>
+                    <version>3.8.0</version>
+                </plugin>
+            </plugins>
+        </pluginManagement>
     </build>
-
-</project>
+</project>
```
Comment (DEz-1996) [inline pom.xml]: Add empty line

File: src/main/java/com/example/springfirsthw/dto/UpdateBookRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+@Data
+public class UpdateBookRequestDto {
+    private String title;
+    private String author;
+    private String isbn;
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
+}
```
Comment (DEz-1996) [inline src/main/java/com/example/springfirsthw/dto/UpdateBookRequestDto.java]: Add empty line

File: src/main/java/com/example/springfirsthw/dto/UpdateBookRequestDto.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package com.example.springfirsthw.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class UpdateBookRequestDto {
```
Comment (DEz-1996) [inline src/main/java/com/example/springfirsthw/dto/UpdateBookRequestDto.java:7]: This Dto is not used anywhere, is it needed here?

**Review by DEz-1996 — APPROVED** [review]
Good job, but have some points to consider

File: src/main/java/com/example/springfirsthw/service/impl/BookServiceImpl.java (hunk 2 above)
Comment (bhdnchui) [reply src/main/java/com/example/springfirsthw/service/impl/BookServiceImpl.java]: Still exist.

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+                    nullable: false
+              - column:
+                  name: isbn
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+                    unique: true
+              - column:
+                  name: price
+                  type: decimal
```
Comment (bhdnchui) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml]:
```suggestion
                  type: decimal(19, 2)
```

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-books-table.yaml
```
Comment (bhdnchui) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: Add an empty line)

**Review by bhdnchui — CHANGES_REQUESTED** [review]
Good job! Just a few points to consider.

File: src/main/java/com/example/springfirsthw/service/impl/BookServiceImpl.java [hunk 9]
```diff
@@ -1,27 +1,51 @@
...
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(NO_BOOK_BY_ID_FOUND_MSG + id));
+        return bookMapper.toBookDto(book);
     }
 
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        getById(id);
```
Comment (bhdnchui) [inline src/main/java/com/example/springfirsthw/service/impl/BookServiceImpl.java]: Let's rewrite it with if statement.
```
if (bookRepository.findById(id).isEmpty()) {
  throw ...
}
```
I know that it is the same check as in `getById` method, but now it looks not obvious.

**Review by bhdnchui — CHANGES_REQUESTED** [review]
Just one point to consider. Also, please pay attention to your commit messages.

**Review by d1sam — APPROVED** [review]
Well done!

**Pull request merged**